src/android/DecryptionExecutor.java

```diff
@@ -0,0 +1,32 @@
+package com.crypho.plugins;
+
+
+import android.util.Log;
+import java.util.concurrent.Callable;
+
+
+public class DecryptionExecutor implements Callable<String> {
+    private static final String TAG = "SecureStorage";
+    private String alias;
+    private byte[] encKey, iv, ct, adata;
+
+    public DecryptionExecutor(byte[] encKey, String alias, byte[] iv, byte[] ct, byte[] adata) {
+        this.encKey = encKey;
+        this.alias = alias;
+        this.iv = iv;
+        this.ct = ct;
+        this.adata = adata;
+    }
+
+    @Override
+    public String call() {
```

Comment:
Maybe is a technical decision. But why not throw the error? Seems like if it does error on decryption you need to check the String for an error message.